packages/twenty-server/src/workspace/messaging/services/fetch-workspace-messages.service.ts

```diff
@@ -224,4 +231,74 @@ export class FetchWorkspaceMessagesService {
       });
     }
   }
+
+  async getAllSavedMessagesIdsForConnectedAccount(
+    dataSourceMetadata: DataSourceEntity,
+    workspaceDataSource: DataSource,
+    connectedAccountId: string,
+  ): Promise<string[]> {
+    const messageIds: { externalId: string }[] =
+      await workspaceDataSource?.query(
+        `SELECT message."externalId" FROM ${dataSourceMetadata.schema}."message" message
+      LEFT JOIN ${dataSourceMetadata.schema}."messageThread" ON message."messageThreadId" = ${dataSourceMetadata.schema}."messageThread"."id"
+      LEFT JOIN ${dataSourceMetadata.schema}."messageChannel" ON ${dataSourceMetadata.schema}."messageThread"."messageChannelId" = ${dataSourceMetadata.schema}."messageChannel"."id"
+      WHERE ${dataSourceMetadata.schema}."messageChannel"."connectedAccountId" = $1`,
+        [connectedAccountId],
+      );
+
+    return messageIds.map((message) => message.externalId);
+  }
+
+  async getAllSavedMessageThreadsIdsForConnectedAccount(
```

Comment:
We can probably get both message and thread external IDs in a single request?

Reply:
Right !